drivers/modem/hl7800.c

```diff
@@ -3943,7 +3945,7 @@ static void sock_read(struct net_buf **buf, uint16_t len)
 		goto exit;
 	}
 
-	if (sock->error) {
+	if (sock->error != 0) {
```

Comment:
Actually previous version `if (sock->error)` seems more readable to me (takes less time to process in my head), as it translates directly to: "is there an error on socket?".